opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -507,7 +507,10 @@ namespace Opm
         }
         this->changed_to_open_this_step_ = false;
         const bool well_operable = this->operability_status_.isOperableAndSolvable();
-        if (!well_operable && old_well_operable) {
+
+        if (this->wellIsStopped() && !changed_to_stopped_this_step_) {
+            // we don't want to reopen stopped wells as they are already open
```

Comment:
I do not have the background related to the targeting situation the PR is trying to solve. But an empty `if` branch does not look good in the code level.

Reply:
I will fix.

Reply:
I agree the parsing of this in my head is hard, assuming it will be improved. But I also do not understand the comment?

Reply:
```c++
if (this->wellIsStopped() && !changed_to_stopped_this_step_) 
```


For this `if` statement to be true, it needs `this->wellIsStopped() ` to be `true` and `changed_to_stopped_this_step_` to be `false`. 

from the code below that updates `changed_to_stopped_this_step_`, it looks like if `this->wellIsStopped() == true`, then `changed_to_stopped_this_step_` will always be `false`, 

Then I will question whether we need to use the `this->wellIsStopped() && !changed_to_stopped_this_step_` together. 

```c++
 512         const bool well_operable = this->operability_status_.isOperableAndSolvable();
 513         if (!well_operable && old_well_operable) {
 514             if (this->well_ecl_.getAutomaticShutIn()) {
 515                 deferred_logger.info(" well " + this->name() + " gets SHUT during iteration ");
 516             } else {
 517                 if (!this->wellIsStopped()) {
 518                     deferred_logger.info(" well " + this->name() + " gets STOPPED during iteration ");
 519                     this->stopWell();
 520                     changed_to_stopped_this_step_ = true;
 521                 }
 522             }
```

Reply:
> from the code below that updates changed_to_stopped_this_step_, it looks like if this->wellIsStopped() == true, then changed_to_stopped_this_step_ will always be false,

I agree there seems to be something strange going on here. I will investigate further. The reason for the extra `if ` was to avoid duplicated "......gets REVIVED during iteration" output. But I agree that even without the extra `if ` the message should not be duplicated. I will remove and see if I am able to reproduce the issue again.

Reply:
I re-checked without the `if ` and there is no difference regarding output. So I just removed it. Thanks for noticing @GitPaean. I guess it covered up some other issue in an earlier version of the code I had.